### DISCUSSION TITLE
How to obtain factors for the virtual site constructions?

### DISCUSSION THREAD

**Discussion by Clare-xiong** [ricalessandri/Martini3-small-molecules#25] [Q&A] [answered]
In the parameter file for caffeine, 

[constraints]
#endif
; i  j  funct length
  1  3   1     0.474 1000000 ; cog
  1  6   1     0.405 1000000 ; cog
  3  6   1     0.599 1000000 ; cog
  3  7   1     0.555 1000000 ; cog
  6  7   1     0.283 1000000 ; cog

[dihedrals]
; improper
; i j k l  funct  ref.angle force_k
  1 3 6 7    2      180       200   

[virtual_sites2]
; sites positioned at 1-a
; site  from  funct    a
   5    3 6     1     0.635 ; cog 

[virtual_sites3]
; peripheral virtual sites
; site positioned as a linear combination of 3 atoms
; the site is in the same plane
; site  from      funct    a      b
   2    5 1 3       1     0.640 0.605 ; cog
   4    5 3 7       1     0.650 0.600 ; cog

In the [virtual_site2], how do we get the factor 'a'? Is factor 'a' equal to 0.38 (i.e., the distance of bead 3 and bead 5) divided by 0.599 (i.e., the distance of bead 3 and bead 6)? And for the factors 'a' and 'b' in the [virtual_site3], I would very much like to know how to calculate them too.

**Comment by ricalessandri** [comment, marked as answer]
Hi @Clare-xiong and thanks for the question. The factors for the virtual sites constructions are obtained in the same way as the bond length and other bonded parameters, i.e., by comparing the mapped distributions coming from all-atom simulations to the CG distributions, as described in [tutorial to parametrize new small-molecules](http://cgmartini.nl/index.php/2021-martini-online-workshop/tutorials/560-5-parametrization-of-a-new-small-molecule) (around section 7)). Namely, in deriving the factors for building the virtual sites, you should consider them as actual particles and hence have bond lengths between the particles and the virtual sites and match such bond lengths (even though you won't have such bond lengths in the final model). Actually, [a chapter we recently published](https://aip.scitation.org/doi/pdf/10.1063/9780735425279_001) describes in quite some detail exactly the making of the caffeine model - see in particular [Fig 1.2a and Fig1.c and associated discussions](https://aip.scitation.org/doi/pdf/10.1063/9780735425279_001).

In principles, given that the virtual sites are built as [linear combinations of the particle positions](https://manual.gromacs.org/documentation/current/reference-manual/functions/interaction-methods.html#virtual-interaction-sites), you can probably devise a cleverer way to obtain those factors. However, you'll still need to extract bond lengths between the real particles as described in the [tutorial to parametrize new small-molecules](http://cgmartini.nl/index.php/2021-martini-online-workshop/tutorials/560-5-parametrization-of-a-new-small-molecule).

**Comment by Clare-xiong** [comment]
I know how to get the factors a and b. Thanks!
